include/ensmallen_bits/de/de.hpp

```diff
@@ -137,6 +138,9 @@ class DE
 
   //! The tolerance for termination.
   double tolerance;
+
+  //! Controls early termination of the optimization process.
+  bool terminate;
```

Comment:
I think the class member is not needed here; if I understand right, it could be done as just a local variable in the `Optimize()` method.  Let me know if I missed something.